src/main/java/de/numcodex/sq2cql/model/Mapping.java

```diff
@@ -7,53 +7,38 @@
 import de.numcodex.sq2cql.model.structured_query.Modifier;
 
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 
+import static java.util.Objects.requireNonNull;
+
 /**
  * @author Alexander Kiel
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
-public final class Mapping {
-
-    private final TermCode key;
-    private final String resourceType;
-    private final String valueFhirPath;
-    private final List<Modifier> fixedCriteria;
-
-    private Mapping(TermCode key, String resourceType, String valueFhirPath, List<Modifier> fixedCriteria) {
-        this.key = Objects.requireNonNull(key);
-        this.resourceType = Objects.requireNonNull(resourceType);
-        this.valueFhirPath = valueFhirPath;
-        this.fixedCriteria = Objects.requireNonNull(fixedCriteria);
+public record Mapping(TermCode key, String resourceType, String valueFhirPath, List<Modifier> fixedCriteria) {
+
+    public Mapping {
+        requireNonNull(key);
+        requireNonNull(resourceType);
+        requireNonNull(valueFhirPath);
+        requireNonNull(fixedCriteria);
     }
 
     public static Mapping of(TermCode concept, String resourceType) {
-        return new Mapping(Objects.requireNonNull(concept), Objects.requireNonNull(resourceType), null, List.of());
+        return new Mapping(requireNonNull(concept), requireNonNull(resourceType), "value", List.of());
     }
 
     @JsonCreator
     public static Mapping of(@JsonProperty("key") TermCode key,
                              @JsonProperty("fhirResourceType") String resourceType,
                              @JsonProperty("valueFhirPath") String valueFhirPath,
                              @JsonProperty("fixedCriteria") Modifier... fixedCriteria) {
-        return new Mapping(Objects.requireNonNull(key), Objects.requireNonNull(resourceType), valueFhirPath,
+        return new Mapping(requireNonNull(key), requireNonNull(resourceType),
```

Comment:
Is requireNonNull here and in line 28 not redundant, as the generated constructor already should enforce nonNull due to lines 20-24?

Reply:
Yes it is. Thanks.